### ISSUE TITLE
IPv6 records missing from FoundryVTT internet services

### ISSUE THREAD

**Issue by felddy** [foundryvtt/foundryvtt#9897]
### What happened?

Users in an IPv6 only environment cannot access FoundryVTT internet services.  e.g., main website, licensing servers, S3 bucket.  This prevents an installation of FoundryVTT from starting up as the license cannot be signed.

There are [work-arounds to use NAT64](https://github.com/felddy/foundryvtt-docker/discussions/752#discussioncomment-6660606), but native IPv6 services would be welcome.

I'm not sure if this a regression or that the `AAAA` records never existed.  I've included a table below of the current availability of `A` and `AAAA` records for the various endpoints that the client accesses.  The table is generated by a [shell script available in this gist](https://gist.github.com/felddy/8021c23a97a5c269e3b0f0a9b25b9cf6). 

## DNS status

| Name  | `A` | `AAAA` | Notes |
|------|:-:|:-----:|-------|
| api.foundryvtt.com | ✅ | ❌ |  |
| foundryvtt.com | ✅ | ❌ |  |
| foundryvtt.s3.amazonaws.com | ✅ | ❌ | `CNAME` to s3-us-west-2-w.amazonaws.com. |
| s3-us-west-2-w.amazonaws.com. | ✅ | ❌ |  |
| foundryvtt.s3.dualstack.us-west-2.amazonaws.com | ✅ | ✅ | `CNAME` to s3-r-w.dualstack.us-west-2.amazonaws.com. |
| s3-r-w.dualstack.us-west-2.amazonaws.com. | ✅ | ✅ |  |

See also:
- https://github.com/felddy/foundryvtt-docker/discussions/752
- https://github.com/felddy/foundryvtt-docker/issues/531
- https://gist.github.com/felddy/8021c23a97a5c269e3b0f0a9b25b9cf6
- @daegalus: https://discord.com/channels/170995199584108546/978002307251863552/1127121221063028806
- [AWS Public IPv4 Address Charge + Public IP Insights](https://aws.amazon.com/blogs/aws/new-aws-public-ipv4-address-charge-public-ip-insights/)

### What ways of accessing Foundry can you encounter this issue in?

- [X] Native App (Electron)
- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [X] Other

### Reproduction Steps

1. Start FoundryVTT `v11.307` in an IPv6-only environment.
1. Note routing errors in logs.
1. Sadness ensues.  

### What core version are you reporting this for?

11.307

### Relevant log output

```shell
# Logs not available at time of issue creation.  
# I will request an update from user @daegalus.
```

### Bug Checklist

- [X] The issue occurs while all Modules are disabled

**Comment by aaclayton** [comment]
Our AWS stack does not, unfortunately, support IPv6 only at this time and will require some significant work to support. This is something we'll prioritize but it's going to be a known issue for a while. It's not as simple as simply adding AAAA records since EC2 does not provide IPv6 addresses

**Comment by daegalus** [comment]
> Our AWS stack does not, unfortunately, support IPv6 only at this time and will require some significant work to support. This is something we'll prioritize but it's going to be a known issue for a while. It's not as simple as simply adding AAAA records since EC2 does not provide IPv6 addresses

No, but it should be just some edits minor configuration edits to the VPC, Route Tables, and EC2 instance.

This guide is from 2020, so the settings might be easier now, but it goes over how to enable IPv6 on existin EC2 instances: https://4sysops.com/archives/assign-an-ipv6-address-to-an-ec2-instance-dual-stack/

If you use ELBs, you can still point to ipv4 only instances, you just need to add ipv6 to the subnet the ELB is in, and enable in ELB.

**Comment by aaclayton** [comment]
> No, but it should be just some edits minor configuration edits to the VPC, Route Tables, and EC2 instance.

I'll look into it - at some point. 

> This guide is from 2020, so the settings might be easier now, but it goes over how to enable IPv6 on existing infrastructure: https://4sysops.com/archives/assign-an-ipv6-address-to-an-ec2-instance-dual-stack/

First sentence "The procedure to assign an IPv6 address to an EC2 in AWS is amazingly complicated." *yep, checks out*

> If you use ELBs, you can still point to ipv4 only instances, you just need to add ipv6 to the subnet the ELB is in, and enable in ELB.

Nope.

> If you use anything like terraform or soemthing, it should be doable through there too, withotu recreating anything.

Also nope.

**Comment by daegalus** [comment]
> > This guide is from 2020, so the settings might be easier now, but it goes over how to enable IPv6 on existing infrastructure: https://4sysops.com/archives/assign-an-ipv6-address-to-an-ec2-instance-dual-stack/
> 
> First sentence "The procedure to assign an IPv6 address to an EC2 in AWS is amazingly complicated." _yep, checks out_

The only thing I would argue is this sentence from the article,. It doesn't check out because the whole thing takes nor more than 5 mins. I linked the article just cause it has all the steps. It's only complicated in the sense that knowing where it all is, but thats what the article covers. the actual changes and why they need changing are really fast. Because it boils down to Enable IPV6 on VPC, add default route ipv6 route entry, Add ipv6 address to EC2 instance, adjust security groups if ipv6 address. Which is just a bit of copy and pasting.

Again, not saying you shouldn't do this when you can spare the time and effort to look into it, i know people and resources are limited. I am just trying to set expectations and help where I can.
